aiida_vasp/workchains/vasp.py

```diff
@@ -176,12 +192,11 @@ def init_inputs(self):
 
         # Perform inputs massage to accommodate generalization in higher lying workchains
         # and set parameters.
-        parameters_massager = ParametersMassage(self, self.inputs.parameters, unsupported_parameters)
+        parameters_massager = ParametersMassage(self, self.ctx.inputs.parameters, unsupported_parameters)
```

Comment:
I do not think `ctx.parameters` is set, but maybe we should and use this to be consistent.

Reply:
To be honest I do not fully remember why I thought that use `ctx.inputs.parameters` was the better choice. Do you think that it would be a good idea to then make sure that everything in the `vasp` workchain uses the `self.ctx.inputs` instead of the bare `self.inputs`?

Reply:
For this case I do not think it matters. If we need to make sure it is the raw input we should use `self.inputs` as that will never be modified. `self.ctx.inputs` might be modified. At least that is what I have been trying to do for the other workchains.

Reply:
I think that we need to have `self.ctx.inputs.parameters` here because the call `self.ctx.inputs.parameters = self._init_parameters()` ensures that all the parameters from the namespaces are inherited, otherwise what would happen is that `dynamics` would not be considered as they are not technically part of the `parameters` dictionary in the raw `self.inputs.parameters`.

Reply:
Yes, I fully support having this and is a consequence of the previous addition (allowing dynamics input on the workchain). Thanks.